Sprint3/form submit api fix

nicolasabousawan opened the pull request:
The API now works with email - which was the recent change based on DB changes ticket

nicolasabousawan committed:
new React Route to Edit Profiles

New react route to edit profile.
Uses existing GET request to pre-populate form. (Waiting on updates to the API to full the form entirely)
PUT insertFormData (to update DB) to be written still

nicolasabousawan committed:
Server side PUT call implemented

PUT call logic set up.

Now we need to connect REACT to LAMBDA

nicolasabousawan committed:
connect dev APIs

connect the dev environment API to dev environment react

nicolasabousawan committed:
Connect dev stage API to react

nicolasabousawan committed:
fix POST createForm API

POST data was out of date due to DB changes... now working

nicolasabousawan commented:
Ignore all other commits except for 66c0e10

The rest are recorded on another PR

nicolasabousawan committed:
Merge branch 'main' into sprint3/formSubmitApiFix

nicolasabousawan commented:
I fixed all conflicts + tool pic is displayable!

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+        this.setState({ref1Email: event.target.value});
+    }
+
+    handleRef2NameChange(event) {
```
I wonder if we could refactor all these methods a bit to take in two args, event and attribute name. That way it would look something like below, and we can delete a ton of this class

```
let handleChange = (event, stateVar) => {
this.setState({stateVar: event.target.value)
}

nicolasabousawan replied:
Good call out - I simply took inspo from Gurleens changes Sprint 2. Should have been more aware. 😬 
Thoughts on making this part of our refactor efforts next sprint? This will need to be done on the createProfile component AND we need to add more fields (which is all work being done next sprint) 
I don't mind doing it now, but it might be worthwhile having 1 refactor ticket for this subset of changes.

kvptkr replied:
Sounds good man, let's push it to next sprint

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+    render() {
+        return (
+            <>
+            <NavBar/>
```
I don't know why, but the NavBar shows up twice for me.
s![shows up twice](https://user-images.githubusercontent.com/16845466/99885031-24907080-2c00-11eb-822e-574578381d4b.png)

 Deleting this NavBar component seemed to solve the issue. 

I'm trying to figure out why, is this a child component of any other components?

nicolasabousawan replied:
Yup, Matt made changes to with the Auth ticket that I did not pull after it was merged in

nicolasabousawan replied:
Thanks for calling it out

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+                        },
+                    })
+                    .then((res) => {
+                        console.log(res);
```
I think that we can safely delete this line/ the then block

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+            <NavBar/>
+            {this.state.loading === (
+            <div>Loading...</div>
+            )}
```
I don't think that this loading logic works properly bro. No matter what the Profile Does not Exist message will be displayed. What you can do is do smt like this:
```
render() {
if (this.state.loading) {
return(
<div><h2>Loading...</h2></div>
)
} else {
return(
//everything else after this
)
}

}
```

Also, instead of the message  "loading...", we could also leverage the brand new loading component that Matt made for his story, it looks p nice.

nicolasabousawan replied:
So this is a semantic error in all our viewProfile component 
It is also because the "loading" variable is not updated after being initialized as well. I would say lets get rid of it and provide a better "loading" experience for Sprint 4 (even though its not that big of a priority imo). For an MVP its functional. 
I am getting rid of the loading, and we should open an issue for the viewProfile component so we can track it

kvptkr replied:
Wait sorry bro, I don't understand. On line 53 after we fetch the data, we do update the loading state to be false, that would kick off a re-render and make the code snippet above work. I think I may be misunderstanding your comment or maybe I wasn't clear w my statement above.

nicolasabousawan replied:
I updated with a new commit - plz see the video I sent you offline

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+    }
+}
+
+export default CreateProfile;
```
newline at EOF

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+
+//Uncomment/Comment based on env
+// const URL = "http://localhost:3001/dev/editProfile";
+const URL = "https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/editProfile";
```
we should change this var name to be like editProfileUrl or something along those lines to be more descriptive

kvptkr commented on ui/src/Components/EditProfile.jsx:
```diff
@@ -0,0 +1,266 @@
+    // function to get new information 
+    async componentDidMount()
+    {
+      const url = `https://bixe448nsa.execute-api.us-west-1.amazonaws.com/dev/viewProfile?id=${this.state.props.match.params.id}`;
```
We should declare this at the top of the file like we do for the first URL to allow for easy switching back and forth bw localhost and aws. Then over here we should just concatenate the `this.state.props.match.params.id`.

kvptkr reviewed (commented):
Left general feedback. Looks really good nick 😄

nicolasabousawan committed:
added view tool pic function

nicolasabousawan committed:
addressing PR feedback

addressing Kanav changes to address potential formatting, maintainability concerns

nicolasabousawan committed:
fixed loading function

leverage new components as transition
